src/bun.js/bindings/bindings.cpp

```diff
@@ -4048,37 +4072,79 @@ class V8StackTraceIterator {
             return false;
         }
 
-        auto firstColon = line.find(':', openingParenthese + 1);
+        auto lineInner = StringView_slice(line, openingParenthese + 1, closingParenthese);
+        auto colonCount = StringView_count(lineInner, ':', 0);
 
-        // if there is no colon, that means we only have a filename and no line number
+        switch (colonCount) {
+        case 0: {
+            // /path/to/file.js
+            frame.sourceURL = lineInner;
+            break;
+        }
 
-        //   at foo (native)
-        if (firstColon == WTF::notFound) {
-            frame.sourceURL = line.substring(openingParenthese + 1, closingParenthese - openingParenthese - 1);
-        } else {
-            // at foo (/path/to/file.js:
-            frame.sourceURL = line.substring(openingParenthese + 1, firstColon - openingParenthese - 1);
+        case 1: {
+            // /path/to/file.js:
+            // /path/to/file.js:1
+            // node:child_process
+            // C:\Users\dave\bun\file.js
+
+            auto marker1 = 0;
+            auto marker2 = lineInner.find(':', marker1);
+            auto marker3 = lineInner.length();
+
+            auto segment1 = StringView_slice(lineInner, marker1, marker2);
+            auto segment2 = StringView_slice(lineInner, marker2 + 1, marker3);
+
+            if (auto int1 = WTF::parseIntegerAllowingTrailingJunk<unsigned int>(segment2)) {
+                frame.sourceURL = segment1;
+                frame.lineNumber = WTF::OrdinalNumber::fromOneBasedInt(int1.value());
+            } else {
+                frame.sourceURL = StringView_slice(lineInner, marker1, marker3);
+            }
+            break;
         }
 
-        if (firstColon != WTF::notFound) {
+        default: {
+            // /path/to/file.js:1:
+            // /path/to/file.js:1:2
+            // node:child_process:1:2
+            // C:\Users\dave\bun\file.js:
+            // C:\Users\dave\bun\file.js:1
+            // C:\Users\dave\bun\file.js:1:2
+
+            auto linenoColon = lineInner.find(':', 0);
+            for (size_t i = 0; i < colonCount - 2; i++) {
+                linenoColon = lineInner.find(':', linenoColon + 1);
```

Comment:
This finds every colon until the 2nd to last colon

That's a lot of duplicate work